### PR TITLE
Refactor MediaPlayer and add CpsAssetMediaPlayer

### LINKED ISSUE #4097: Add AV block to MAPs
**Is your feature request related to a problem? Please describe.**
We need to add a preprocessor rule that takes an `aresMedia` block and generates an Optimo format AV block/s. 

**Describe the solution you'd like**
Create a new generator that takes an `aresMedia` block and creates an Optimo format `video` block with support for `caption`, `aresMediaMetadata`, `aresMedia`, `image` etc. This should then be rendered using the blocks container logic similar to how Articles render the Media Player component.

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Snapshot testing and E2Es are required to ensure a headline is rendered on the page.

- [ ] This feature is expected to need manual testing.

**Additional context**
- [Articles example with AV](https://www.test.bbc.co.uk/news/articles/c3wmq4d1y3wo.json)
- [CPS example with AV](https://ares-broker.api.bbci.co.uk/api/asset/pidgin/media-49189067)

Optimo format video block:
```
        {
          "model": {
            "blocks": [
              {
                "model": {
                  "blocks": [
                    {
                      "model": {
                        "blocks": [
                          {
                            "model": {
                              "blocks": [
                                {
                                  "model": {
                                    "attributes": [
                                      
                                    ],
                                    "text": "Video, Clip, International and UK, Guidance, Subtitles"
                                  },
                                  "type": "fragment"
                                }
                              ],
                              "text": "Video, Clip, International and UK, Guidance, Subtitles"
                            },
                            "type": "paragraph"
                          }
                        ]
                      },
                      "type": "text"
                    }
                  ]
                },
                "type": "caption"
              },
              {
                "model": {
                  "blocks": [
                    {
                      "blockId": "urn:bbc:ares::clip:p01k6msm",
                      "model": {
                        "advertising": true,
                        "embedding": true,
                        "format": "audio_video",
                        "id": "p01k6msm",
                        "imageCopyright": "BBC",
                        "imageUrl": "ichef.test.bbci.co.uk/images/ic/$recipe/p01k6mtv.jpg",
                        "subType": "clip",
                        "syndication": {
                          "destinations": [
                            
                          ]
                        },
                        "synopses": {
                          "short": "They may be tiny, but us humans could learn a thing or two from ants."
                        },
                        "title": "Five things ants can teach us about management",
                        "versions": [
                          {
                            "availableFrom": 1540218932000,
                            "availableTerritories": {
                              "nonUk": true,
                              "uk": true
                            },
                            "duration": 191,
                            "durationISO8601": "PT3M11S",
                            "types": [
                              "Original"
                            ],
                            "versionId": "p01k6msp",
                            "warnings": {
                              "long": "Contains strong language and adult humour.",
                              "short": "Contains strong language and adult humour."
                            }
                          }
                        ]
                      },
                      "type": "aresMediaMetadata"
                    },
                    {
                      "model": {
                        "blocks": [
                          {
                            "model": {
                              "copyrightHolder": "BBC",
                              "height": 1080,
                              "locator": "ichef.test.bbci.co.uk/images/ic/$widthxn/p01k6mtv.jpg",
                              "originCode": "mpv",
                              "width": 1920
                            },
                            "type": "rawImage"
                          },
                          {
                            "model": {
                              "blocks": [
                                {
                                  "model": {
                                    "blocks": [
                                      {
                                        "model": {
                                          "blocks": [
                                            {
                                              "model": {
                                                "attributes": [
                                                  
                                                ],
                                                "text": "Ants"
                                              },
                                              "type": "fragment"
                                            }
                                          ],
                                          "text": "Ants"
                                        },
                                        "type": "paragraph"
                                      }
                                    ]
                                  },
                                  "type": "text"
                                }
                              ]
                            },
                            "type": "altText"
                          }
                        ]
                      },
                      "type": "image"
                    }
                  ]
                },
                "type": "aresMedia"
              }
            ],
            "locator": "urn:bbc:pips:pid:p01k6msm"
          },
          "type": "video"
        },
```

### LINKED ISSUE #4418: Refactor media player to avoid override logic
**Is your feature request related to a problem? Please describe.**
In https://github.com/bbc/simorgh/pull/4410 override logic was added to articles, ideally this should be changed by splitting the containers into sub containers.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by dr3**
Resolves #4097
Resolves #4418

**Overall change:** _A very high-level summary of easily-reproducible changes that can be understood by non-devs._

Article: http://localhost:7080/news/articles/c3wmq4d1y3wo
STY: http://localhost:7080/igbo/afirika-23252735
MAP: http://localhost:7080/pidgin/23248703

**Code changes:**

- Extracts variable logic into parent containers for articles and cps assets
- Adds a media player container for MAP pages
- Adds cypress tests for MAP media player

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

**Comment by ryanmccombe** [comment]
I’m not too sure why we need a wrapper just to pass props to a child component - why not pass the props?

If it’s because the blocks container doesn’t allow that, we can remove that problem: https://github.com/bbc/simorgh/issues/4417

I didn’t want to modify the blocks / articles container for the initial increment, but if we’re doing that anyway, the direction I was going (and what I’d still suggest) is something like this:

```js
// Articles Page
const componentsToRender = {
   video: <MediaPlayer assetId={assetId} showPlaceholder />
}

<Blocks components={componentsToRender} />

// MAP
const componentsToRender = {
  video: <MediaPlayer assetId={assetId} wrapper={GridItemWide} />
}

<Blocks components={componentsToRender} />
```
Or any similar variation:
```js
<MediaPlayer getEmbedUrl={someFunc} columns={8} />
<MediaPlayer getEmbedUrl={someFunc} wide />
<MediaPlayer getEmbedUrl={someFunc}>
  {iframe => <GridItemLarge>{iframe}</GridItemLarge>
</MediaPlayer>
```

Making our existing code more flexible seems a better strategy than adding more code

**Comment by dr3** [comment]
> I’m not too sure why we need a wrapper just to pass props to a child component - why not pass the props?

@ryanmccombe personally i'm not a fan of cluttered containers like that, I had to do it slightly, as seen below, but where its avoidable we should 100% move containers into their own file if possible

```jsx
const componentsToRender = {
    headline: headings,
    text,
    image,
    timestamp,
    video: props => <MediaPlayer {...props} assetUri={assetUri} />,
  };
```

If I could change above to just have `video: mediaPlayer` I would, however this would likely require some context work to provide it with `assetUri`

**Comment by ryanmccombe** [comment]
You can implement this deviation without adding any new components, or any clutter to existing components

[This suggestion](https://gist.github.com/ryanmccombe/69db8a6fd9b3e6eeccfb1a217c7fc4a2) removes all 3 containers, including the one you're not happy about above.

File: src/app/containers/ArticleMediaPlayer/index.test.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import {
+  validAresMediaVideoBlock,
+  defaultToggles,
+} from '../MediaPlayer/fixtureData';
+
+const GenerateMediaPlayer = ({
+  /* eslint-disable react/prop-types */
+  platform,
+  blocks,
+  /* eslint-disable react/prop-types */
```
Comment (ryanmccombe) [inline src/app/containers/ArticleMediaPlayer/index.test.jsx]: Does this have any effect?

File: src/app/containers/CpsAssetMediaPlayer/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  validAresMediaVideoBlock,
+  defaultToggles,
+} from '../MediaPlayer/fixtureData';
+
+const GenerateMediaPlayer = ({
+  /* eslint-disable react/prop-types */
+  platform,
+  blocks,
+  assetUri,
+  /* eslint-disable react/prop-types */
```
Comment (ryanmccombe) [inline src/app/containers/CpsAssetMediaPlayer/index.test.jsx]: Does this syntax disable eslint for a specific block of code?  Haven't seen it before

File: src/app/containers/ArticleMediaPlayer/index.test.jsx (hunk 1 above)
Comment (dr3) [reply src/app/containers/ArticleMediaPlayer/index.test.jsx]: Ahh the second one should be an `enable` 🙈

File: src/app/containers/CpsAssetMediaPlayer/index.test.jsx (hunk 2 above)
Comment (dr3) [reply src/app/containers/CpsAssetMediaPlayer/index.test.jsx]: It does when the second one is `enable`, whoops, fixed :)

**Review by ryanmccombe — APPROVED** [review]

**Review by jamesdonoh — APPROVED** [review]
Happy with the refactor, although the new Cypress tests feel out of place on this PR no?

Please could you add some brief context/rationale for the change to the description too?

**Comment by LilyL0u** [comment]
Video playability and positioning is the same as on latest (but also now shows as media players rather than just images). On MAPs and Articles. Article page screenshots below:

Articles:

Latest
Image 01-11-2019 at 16 23

This branch:
Image 01-11-2019 at 17 38

Differentfint showing but this is not part of the test for this change.

STY page link still leads to a page with the player not working but this is still yet to be fixed and is broken on both latest and this branch.

UNADJUSTEDNONRAW_thumb_71

All ok @dr3 ?

**Comment by dr3** [comment]
@LilyL0u Perfect :)

**Pull request merged**